### PR TITLE
Disk persistence

### PR DESCRIPTION
**Pull request by priyankaranke**
Disk persistence implementation and tests pushed.

### CODE REVIEW

File: project/src/key_value_store.cpp [hunk 1]
```diff
@@ -1,5 +1,68 @@
...
+
+// populate map_ from file if file is given. Error if file is invalid
+KvStore::KvStore(const std::string &file_str) : filename(file_str) {
+  // no kv backing file given
+  if (file_str == "") {
+    return;
+  }
+
+  std::ifstream infile(filename);
+  if (!infile.good()) {
```
Comment (RaymondJune) [inline project/src/key_value_store.cpp:21]: probably add a log message here

File: project/src/key_value_store_main.cpp [hunk 2]
```diff
@@ -17,6 +38,7 @@ void RunServer() {
 int main(int argc, char** argv) {
   // Initialize Google's logging library.
   google::InitGoogleLogging(argv[0]);
+  gflags::ParseCommandLineFlags(&argc, &argv, true);
   LOG(INFO) << "Starting Key Value Store Server " << std::endl;
   RunServer();
   return 0;
```
Comment (RaymondJune) [inline project/src/key_value_store_main.cpp:44]: shutdown arg parser

File: project/src/key_value_store.cpp [hunk 3]
```diff
@@ -1,5 +1,68 @@
 #include "key_value_store.h"
+#include <glog/logging.h>
+#include <fstream>
 #include <iostream>
+#include <regex>
```
Comment (RaymondJune) [inline project/src/key_value_store.cpp:5]: module not used

File: project/src/key_value_store.cpp (hunk 1 above)
Comment (priyankaranke) [reply project/src/key_value_store.cpp:21]: fixed

File: project/src/key_value_store_main.cpp (hunk 2 above)
Comment (priyankaranke) [reply project/src/key_value_store_main.cpp:44]: fixed

File: project/src/key_value_store.cpp (hunk 3 above)
Comment (priyankaranke) [reply project/src/key_value_store.cpp:5]: fixed

**Comment by priyankaranke** [comment]
Thanks for your review. I have made all the request changes

**Pull request merged**